scripts/cve_check.py

```diff
@@ -0,0 +1,642 @@
+
+    cves = check_kernel_cves_by_cip_kernel_sec(uniq_installed_pkgs, cves, kernel_src_dir, cip_kernel_sec_dir, db_file)
+    return cves
+
+def write_text(cves, output_dir, uniq_installed_pkgs):
+    for pkgname in cves:
+        filename = f"{output_dir}/{pkgname}"
+
+        if len(cves[pkgname]) == 0:
+            continue
+
+        with open(filename, "w") as f:
+            for cve in sorted(cves[pkgname]):
+                info = cves[pkgname][cve]
+                f.write(f"PACKAGE NAME: {info['PACKAGE NAME']}\n")
+                f.write(f"BINARY PACKAGE NAME: {' '.join(info['BINARY PACKAGE NAME'])}\n")
+                f.write(f"VERSION: {info['VERSION']}\n")
+                f.write(f"CVE: {info['CVE']}\n")
+                f.write(f"CVE STATUS: {info['CVE STATUS']}\n")
+                f.write(f"CVE SUMMARY: {info['CVE SUMMARY']}\n")
+                f.write(f"CVSS v2 BASE SCORE: {info['CVSS v2 BASE SCORE']}\n")
+                f.write(f"CVSS v3 BASE SCORE: {info['CVSS v3 BASE SCORE']}\n")
+                f.write(f"VECTOR: {info['VECTOR']}\n")
+                f.write(f"VECTOR STRING: {info['VECTOR STRING']}\n")
+                f.write(f"MORE INFORMATION: {info['MORE INFORMATION']}\n")
+                f.write("\n")
+
+def write_json(cves, output_dir, uniq_installed_pkgs, cve_products):
+    filenames = []
+
+    for pkgname in cves:
+        filename = f"{output_dir}/{pkgname}_cve.json"
+        filenames.append(filename)
+
+        info = cves[pkgname]
+        pkginfo = uniq_installed_pkgs[pkgname]
+
+        product = pkgname
+        if pkgname in cve_products:
+            product = cve_products[pkgname]["product"]
+
+        cvesInRecord = "Yes"
+        if len(info) == 0:
+            cvesInRecord = "No"
+            issues = []
+        else:
+            issues = []
+            for cve in sorted(cves[pkgname]):
+                issues.append(cves[pkgname][cve])
+
+        data = {
+            "version": "1",
+            "package": [
+                {
+                    "name": pkginfo["source"],
+                    "binary package name": pkginfo["bin_pkgs"],
+                    "version": str(pkginfo["version"]),
+                    "products": [
+                        {
+                            "product": product,
+                            "cvesInRecord": cvesInRecord,
+                        },
+                    ],
+                    "issue": issues,
+                },
+            ],
+        }
+
+        with open(filename, "w") as f:
+            json.dump(data, f, indent=4, sort_keys=False)
+
+    return sorted(filenames)
+
+def write_all_one_json(output_dir, image_name, json_filenames):
+    all_in_one_data = {
+        "version": "1",
+        "package": [],
+    }
+    
+    for filename in json_filenames:
+        with open(filename, "r") as f:
+            data = json.load(f)
+            all_in_one_data["package"].append(data["package"])
```

Comment:
The data["package"] is an array, so all_in_one_data["package"] becomes an array of arrays. We should need only elements of the data["package"].

Reply:
fixed.